docs/cookbook/creating-custom-schema-fields.md

```diff
@@ -0,0 +1,999 @@
+
+The `_super` argument is passing the original function (you can read more about this pattern in the [documentation](/reference/module-api/module-overview.html#the-extension-pattern)). The original method takes a single argument, `req`, that is passed as the second argument.
+
+The `browserData` variable is set to the original value of `req` by invoking the original method with the `req` argument. That object is then updated with the custom schema array and the new `req` returned, just like the original method.
+
+### Creating the Vue component
+
+The custom field browser-side code should be added to your module's `ui/apos/components` folder. The file name should match the value passed to the `self.apos.schema.addFieldType()` method `vueComponent` property.
+
+<AposCodeBlock>
+
+```javascript
+<template>
+  <AposInputWrapper
+  :field="field"
+  :error="null"
+  :uid="uid"
+  :modifiers="modifiers"
+  >
+    <template #body>
+      <div class="custom-input-object">
+        <div class="custom-input-wrapper">
+          <AposSchema
+          :schema="gradeSchema"
+          :trigger-validation="triggerValidation"
+          :utility-rail="false"
+          :generation="generation"
+          :conditions-met="conditionsMet"
+          :following-values="followingValues"
+          :disabled="disabled"
+          v-model="gradeSchemaInput">
+          </AposSchema>
+        </div>
+      </div>
+    </template>
+  </AposInputWrapper>
+</template>
+
+<script>
+import AposInputMixin from 'apostrophe/modules/@apostrophecms/schema/ui/apos/mixins/AposInputMixin';
+import AposInputChoicesMixin from 'apostrophe/modules/@apostrophecms/schema/ui/apos/mixins/AposInputChoicesMixin';
+import AposInputWrapper from 'apostrophe/modules/@apostrophecms/schema/ui/apos/components/AposInputWrapper.vue';
+import AposSchema from 'apostrophe/modules/@apostrophecms/schema/ui/apos/components/AposSchema.vue';
+
+export default {
+  name: 'InputGradeField',
+  mixins: [AposInputMixin, AposInputChoicesMixin],
+  components: {
+    AposInputWrapper,
+    AposSchema
+  },
+  props: {
+    generation: {
+      type: Number,
+      required: false,
+      default() {
+        return null;
+      }
+    }
+  },
+  data() {
+    const next = this.getNext();
+    const gradeSchema = apos.modules['grade'].gradeSchema;
+    return {
+      next,
+      gradeSchemaInput: {
+        data: next
+      },
+      gradeSchema: gradeSchema
+    };
+  },
+  watch: {
+    generation() {
+      this.next = this.getNext();
+      this.gradeSchemaInput = {
+        data: this.next.value
+      };
+    },
+    gradeSchemaInput: {
+      deep: true,
+      handler() {
+        if (!this.gradeSchemaInput.hasErrors) {
+          this.next = this.gradeSchemaInput.data;
+        };
+      }
+    }
+  },
+  methods: {
+    getNext() {
+      return this.value.data ? this.value.data : (this.field.def || {});
+    },
+    validate(value) {
+      if (value == null) {
```

Comment:
The subfields are already required, so in effect this field type is always required, which the third if statement will enforce. That makes the first two if statements removable.

Reply:
First if statement removed.